yoti_python_sdk/date_parser.py

```diff
@@ -9,4 +14,29 @@ def from_iso_format(string):
     if len(parts) != 3:
         raise ValueError
 
-    return datetime.date(parts[0], parts[1], parts[2])
+    return date(parts[0], parts[1], parts[2])
+
+
+def datetime_with_microsecond(string):
+    # Python2 does not have a way of parsing date formats.
+    # TODO Deprecate this once Python2 support is dropped.
+    time_split = re.split("[^0-9]", string)
+    parts = len(time_split)
+    if parts <= 6:
+        if logging.getLogger().propagate:
+            logging.warning(ERROR_PARSING_DATE)
+        return None
+
+    year = int(time_split[0]) if parts > 0 else 0
+    month = int(time_split[1]) if parts > 1 else 0
+    day = int(time_split[2]) if parts > 2 else 0
+    hour = int(time_split[3]) if parts > 3 else 0
+    minute = int(time_split[4]) if parts > 4 else 0
+    second = int(time_split[5]) if parts > 5 else 0
+    microsecond = int(float("0." + time_split[6]) * 1e6) if parts > 6 else 0
+    try:
+        return datetime(year, month, day, hour, minute, second, microsecond)
+    except ValueError:
+        if logging.getLogger().propagate:
+            logging.warning(ERROR_PARSING_DATE)
+        return None
```

Comment:
We need to cover every branch in our code, even when it's an interim solution like this for when we remove Python 2.7 support. 

So because this is not a simple function, can we add tests that cover every possible scenario here? Including:
1. less than 6 parts (but not empty string)
1. that the other parts are set to 0 for each of the differing number of parts
1. That ValueError is caught
1. That other errors are propagated

(Apologies if there are some tests that I'm missing)